File: Source/TS/game/tile.ts
import { Vector2 } from "../utilities/Vector2.js";

export abstract class Tile
{
    // Settings
    readonly position: Vector2;
    traversable: boolean;

    constructor(position: Vector2, traversable: boolean = true)
    {
        this.position = position.clone();
        this.traversable = traversable;
    }

    abstract update(deltaTime: number): void;
}

File: core/src/main/java/io/github/RimIsland/Game/Tile.java
package io.github.RimIsland.Game;

import com.badlogic.gdx.math.Vector2;

public abstract class Tile
{
     public Vector2 position;
     public boolean traversable = true;

     public Tile(Vector2 position)
     {
         this.position = position;
     }
}
